File: src/app/gkill/api/req_res/update_nlog_request.go
package req_res

import "github.com/mt3hr/gkill/src/app/gkill/dao/reps"

type UpdateNlogRequest struct {
	SessionID string `json:"session_id"`

	Nlog *reps.Nlog `json:"nlog"`

	TXID *string `json:"tx_id"`

	LocaleName string `json:"locale_name"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`

	WantResponseKyou bool `json:"want_response_kyou"`
}

File: src/client/classes/api/req_res/update-nlog-request.ts
'use strict'

import { Nlog } from '@/classes/datas/nlog'
import { GkillAPIRequest } from '../gkill-api-request'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateNlogRequest extends GkillAPIRequest {

    nlog: Nlog

    tx_id: string | null = null

    want_response_kyou: boolean

    updated_kyou: Kyou | null | null = null

    constructor() {
        super()
        this.nlog = new Nlog()
        this.want_response_kyou = false
    }

}


